### ISSUE TITLE
Single speed pumps are not turned off

### ISSUE THREAD

**Issue by ljscz** [garbled1/pybalboa#7]
The latest code is not not able to turn off any of my pumps. Turning them on works fine. All of them have just one speed s just one press of the button is needed. When the iteration was set to 1 in such case before it worked fine. 

```
       # toggle until we hit the desired state
        for i in range(0, (newstate-self.pump_status[pump]) % (self.pump_array[pump]+1)):
            await self.send_message(*mtypes[BMTS_CONTROL_REQ], C_PUMP1 + pump, 0x00)
            await asyncio.sleep(1.0)
```

**Comment by natekspencer** [comment]
Can you print `self.pump_array` and `self.pump_status` when the pump is both on/off? I tried to code this around a variable number of speeds where a pump with only on/off has a status of 0/1, a pump with on/low/high has a status of 0/1/2, and one with on/low/medium/high would have 0/1/2/3. I'm wondering, though, if it reports differently for on on single speed pumps.

**Comment by natekspencer** [comment]
@ljscz if you wanna pull the latest from my branch into your environment, it guarantees that a minimum of one button push will happen to toggle the pump.

**Comment by ljscz** [comment]
I see following results:

When turning on (spa.PUMP_LOW as spa.PUMP_HIGH does not work)

```
Current Status: Off
self.pump_array
1 1 0 0 0 0
self.pump_status
0 0 0 0 0 0
newstate, self.pump_status[pump], self.pump_array[pump]: 1, 0, 1
```

When running and trying to turn off
```
Current Status: High
self.pump_array
1 1 0 0 0 0
self.pump_status
2 0 0 0 0 0
newstate, self.pump_status[pump], self.pump_array[pump]: 0, 2, 1
```

So it seems that it turn on LOW but then reports HIGH...

@natekspencer Your fix works for me in my configuration. Thanks.

**Comment by natekspencer** [comment]
Ok cool, that's good to know that it reports 0/2 for off/on(high) rather than 0/1 for off/on(low). I wonder if other pumps show differently as well.

**Comment by garbled1** [comment]
Looks like nate fixed this one, I approved the PR

**Comment by garbled1** [comment]
Fixed in 0.13
